services/venice-controller/src/main/java/com/linkedin/venice/controller/stats/LogCompactionStats.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.logging.log4j.Logger;
+
+
+public class LogCompactionStats extends AbstractVeniceStats {
+  private final boolean emitOpenTelemetryMetrics;
+  private final VeniceOpenTelemetryMetricsRepository otelRepository;
+  private final Map<VeniceMetricsDimensions, String> baseDimensionsMap;
+  private static final Logger LOGGER = LogManager.getLogger(LogCompactionStats.class);
+
+  /** metrics */
+  private final MetricEntityStateGeneric repushCallCountMetric;
+  private final MetricEntityStateGeneric compactionEligibleMetric;
+  private final MetricEntityStateGeneric storeNominatedForCompactionCountMetric;
+
+  public LogCompactionStats(MetricsRepository metricsRepository, String clusterName) {
+    super(metricsRepository, "LogCompactionStats");
+    if (metricsRepository instanceof VeniceMetricsRepository) {
+      VeniceMetricsRepository veniceMetricsRepository = (VeniceMetricsRepository) metricsRepository;
+      VeniceMetricsConfig veniceMetricsConfig = veniceMetricsRepository.getVeniceMetricsConfig();
+      emitOpenTelemetryMetrics = veniceMetricsConfig.emitOtelMetrics();
+      if (emitOpenTelemetryMetrics) {
+        this.otelRepository = veniceMetricsRepository.getOpenTelemetryMetricsRepository();
+        this.baseDimensionsMap = new HashMap<>();
+        this.baseDimensionsMap.put(VENICE_CLUSTER_NAME, clusterName);
+      } else {
+        this.otelRepository = null;
+        this.baseDimensionsMap = new HashMap<>();
+      }
+    } else {
+      this.emitOpenTelemetryMetrics = false;
+      this.otelRepository = null;
+      this.baseDimensionsMap = new HashMap<>();
+    }
+
+    repushCallCountMetric = MetricEntityStateGeneric.create(
+        ControllerMetricEntity.REPUSH_CALL_COUNT.getMetricEntity(),
+        otelRepository,
+        this::registerSensor,
+        ControllerTehutiMetricNameEnum.REPUSH_CALL_COUNT,
+        Collections.singletonList(new Count()),
+        baseDimensionsMap);
+
+    compactionEligibleMetric = MetricEntityStateGeneric.create(
+        ControllerMetricEntity.COMPACTION_ELIGIBLE_STATE.getMetricEntity(),
+        otelRepository,
+        this::registerSensor,
+        ControllerTehutiMetricNameEnum.COMPACTION_ELIGIBLE_STATE,
+        Collections.singletonList(new Count()),
+        baseDimensionsMap);
+
+    storeNominatedForCompactionCountMetric = MetricEntityStateGeneric.create(
+        ControllerMetricEntity.STORE_NOMINATED_FOR_COMPACTION_COUNT.getMetricEntity(),
+        otelRepository,
+        this::registerSensor,
+        ControllerTehutiMetricNameEnum.STORE_NOMINATED_FOR_COMPACTION_COUNT,
+        Collections.singletonList(new Count()),
+        baseDimensionsMap);
+  }
+
+  public void recordRepushStoreCall(
+      String storeName,
+      RepushStoreTriggerSource triggerSource,
+      VeniceResponseStatusCategory executionStatus) {
+    if (!this.emitOpenTelemetryMetrics) {
+      LOGGER.warn("Skipping recordRepushStoreCall, emitOpenTelemetryMetrics is false");
```

Comment:
lets remove this check and the log. Tehuti metric should be emitted regardless of whether otel is enabled or not.